Documentation/api-reference.md

```diff
@@ -0,0 +1,374 @@
+| [resetIdentities](#resetIdentities)            |
+| [updateIdentities](#updateIdentities)          |
+
+
+### extensionVersion
+
+The extensionVersion() API returns the version of the Identity for Edge Network extension.
+
+**Syntax**
+
+```swift
+static var extensionVersion: String
+```
+**Examples**
+
+```swift
+let extensionVersion = EdgeIdentity.extensionVersion
+```
+
+### getExperienceCloudId
+
+This API retrieves the Experience Cloud ID (ECID) that was generated when the app was initially launched. This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall.
+
+**Syntax**
+
+```swift
+static func getExperienceCloudId(completion: @escaping (String?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the ECID is available.  The default timeout is 1000ms.
+
+**Examples**
+
+```swift
+Identity.getExperienceCloudId { (ecid, error) in
+  if let error = error {
+    // handle error here
+  } else {
+    // handle the retrieved ID here
+  }
+}
+```
+
+### getIdentities
+
+Get all the identities in the Identity for Edge Network extension, including customer identifiers which were previously added.
+
+**Syntax**
+
+```swift
+static func getIdentities(completion: @escaping (IdentityMap?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the identities are available.  The default timeout is 1000ms. The return format is an instance of [IdentityMap](#identitymap).
+
+**Examples**
+
+```swift
+Identity.getIdentities { (identityMap, error) in
+  if let error = error {
+    // handle error here
+  } else {
+    // handle the retrieved identities here
+  }
+}
+```
+
+### getUrlVariables
+{% hint style="info" %}
+This method was added in Edge Identity version 2.0.0.
+{% endhint %}
+
+
+This API returns the identifiers in URL query parameter format for consumption in **hybrid mobile applications**. There is no leading & or ? punctuation as the caller is responsible for placing the variables in their resulting URL in the correct locations. If an error occurs while retrieving the URL variables, the completion handler is called with a nil value and AEPError instance. Otherwise, the encoded string is returned, for example: `"adobe_mc=TS%3DTIMESTAMP_VALUE%7CMCMID%3DYOUR_ECID%7CMCORGID%3D9YOUR_EXPERIENCE_CLOUD_ID"`
+
+* The `adobe_mc` attribute is an URL encoded list that contains:
+  * `MCMID` - Experience Cloud ID \(ECID\)
+  * `MCORGID` - Experience Cloud Org ID
+  * `TS` - A timestamp taken when this request was made
+
+**Syntax**
+
+```swift
+static func getUrlVariables(completion: @escaping (String?, Error?) -> Void)
+```
+* _completion_ is invoked after the url variables string is available. The default timeout is 1000ms.
+
+**Examples**
+
+```swift
+Identity.getUrlVariables { (urlVariables, error) in
+  if let error = error {
+    // handle error here
+  } else {
+    // handle the retrieved urlVariables encoded string here
+  }
+}
```

Comment:
Would be helpful to have a fully baked example where the urlVariables are attached to an existing url and it is opened in a web view.